central/complianceoperator/v2/profiles/datastore/datastore_impl.go

```diff
@@ -3,21 +3,66 @@ package datastore
 import (
 	"context"
 
+	edge "github.com/stackrox/rox/central/complianceoperator/v2/profiles/profileclusteredge/store/postgres"
 	"github.com/stackrox/rox/central/complianceoperator/v2/profiles/store/postgres"
 	"github.com/stackrox/rox/generated/storage"
-	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+var (
+	complianceOperatorSAC = sac.ForResource(resources.ComplianceOperator)
 )
 
 type datastoreImpl struct {
-	store postgres.Store
+	store            postgres.Store
+	profileEdgeStore edge.Store
 }
 
 // UpsertProfile adds the profile to the database
-func (d *datastoreImpl) UpsertProfile(_ context.Context, _ *storage.ComplianceOperatorProfileV2) error {
-	return errox.NotImplemented
+func (d *datastoreImpl) UpsertProfile(ctx context.Context, profile *storage.ComplianceOperatorProfileV2, clusterID string, profileUID string) error {
+	if ok, err := complianceOperatorSAC.WriteAllowed(ctx); err != nil {
+		return err
+	} else if !ok {
+		return sac.ErrResourceAccessDenied
+	}
+
+	if err := d.store.Upsert(ctx, profile); err != nil {
+		return err
+	}
+
+	profileEdge := &storage.ComplianceOperatorProfileClusterEdge{
+		Id:         uuid.NewV4().String(),
+		ProfileId:  profile.GetId(),
+		ProfileUid: profileUID,
+		ClusterId:  clusterID,
+	}
+	return d.profileEdgeStore.Upsert(ctx, profileEdge)
```

Comment:
The profile is the profile.  So that should be OK (though not really semantically correct).  It would impact that specific cluster when pulling the profiles.  That should resolve itself the next sync request.  I am going to write a ticket though to ensure the sync with sensor is working appropriately.  I'm not terribly familiar with that so I would like to separate it from this in order to get moving on the API side of profiles and to dedicate the proper amount of time to that sync process.  At the same time, I'll wrap this in a transaction now.

https://issues.redhat.com/browse/ROX-21147